### PR TITLE
Treat internal vm errors as internal rpc errors

### PR DESCRIPTION
**Pull request by joshklop**
See https://github.com/NethermindEth/juno/pull/1578#discussion_r1432860435

### CODE REVIEW

**Comment by omerfirmak** [comment]
pushed a simple refactor, let me know if it looks good to you.

File: rpc/handlers.go [hunk 1]
```diff
@@ -1633,12 +1626,9 @@ func (h *Handler) traceBlockTransactions(ctx context.Context, block *core.Block,
...
 		sequencerAddress, state, h.network, paidFeesOnL1, false, false, false, block.Header.GasPrice, block.Header.GasPriceSTRK, legacyJSON)
 	if err != nil {
-		if isVMInternalErr(err) {
-			return nil, ErrInternal.CloneWithData(err.Error())
-		}
 		// Since we are tracing an existing block, we know that there should be no errors during execution. If we encounter any,
-		// report them as unexpected errors
-		return nil, ErrUnexpectedError.CloneWithData(err.Error())
+		// report them as internal errors
+		return nil, ErrInternal.CloneWithData(err.Error())
```
Comment (joshklop) [inline rpc/handlers.go:1631]: Are we reverting the decision made [before](https://github.com/NethermindEth/juno/pull/1475#discussion_r1407919925)? If so, we can remove `vm.ErrInternal` entirely.

Comment (omerfirmak) [reply rpc/handlers.go:1631]: I think we can, since we only count internal errors as failed requests now.

Comment (omerfirmak) [reply rpc/handlers.go:1631]: > we can remove vm.ErrInternal entirely.

Yeap, let's do that as well

**Pull request merged**